subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesDuringTheBuildFileSystemWatchingIntegrationTest.groovy

```diff
@@ -102,47 +136,91 @@ class ChangesDuringTheBuildFileSystemWatchingIntegrationTest extends AbstractFil
 
         when:
         inputFile.text = "initial"
-        runWithFileSystemWatchingAndMakeChangesWhen("waitForUserChanges", "userInput") {
+        runWithRetentionAndDoChangesWhen("waitForUserChanges", "userInput") {
             inputFile.text = "changed"
             waitForChangesToBePickedUp()
         }
         then:
         executedAndNotSkipped(":consumer")
         outputFile.text == "initial"
-        vfsLogs.receivedFileSystemEventsInCurrentBuild >= 0
-        vfsLogs.retainedFilesInCurrentBuild > 0
+        projectFilesInVfs == 1
 
         when:
-        runWithFileSystemWatchingAndMakeChangesWhen("waitForUserChanges", "userInput") {
+        runWithRetentionAndDoChangesWhen("waitForUserChanges", "userInput") {
             inputFile.text = "changedAgain"
             waitForChangesToBePickedUp()
         }
         then:
         executedAndNotSkipped(":consumer")
         outputFile.text == "changed"
-        vfsLogs.receivedFileSystemEventsSinceLastBuild >= 0
         vfsLogs.receivedFileSystemEventsInCurrentBuild >= 1
-        vfsLogs.retainedFilesSinceLastBuild > 0
-        vfsLogs.retainedFilesInCurrentBuild > 0
+        projectFilesInVfs == 1
 
         when:
         server.expect("userInput")
         withWatchFs().run("waitForUserChanges")
         then:
         executedAndNotSkipped(":consumer")
         outputFile.text == "changedAgain"
-        vfsLogs.receivedFileSystemEventsSinceLastBuild >= 0
-        vfsLogs.receivedFileSystemEventsInCurrentBuild >= 0
-        vfsLogs.retainedFilesSinceLastBuild > 0
-        vfsLogs.retainedFilesInCurrentBuild > 0
+        projectFilesInVfs == 2
     }
 
-    private void runWithFileSystemWatchingAndMakeChangesWhen(String task, String expectedCall, Closure action) {
+    private void runWithRetentionAndDoChangesWhen(String task, String expectedCall, Closure action) {
         def handle = withWatchFs().executer.withTasks(task).start()
         def userInput = server.expectAndBlock(expectedCall)
         userInput.waitForAllPendingCalls()
         action()
         userInput.releaseAll()
         result = handle.waitForFinish()
     }
+
+    int getProjectFilesInVfs() {
+        def retainedInformation = result.getOutputLineThatContains("Project files in VFS: ")
+        def numberMatcher = retainedInformation =~ /Project files in VFS: (\d+)/
+        return numberMatcher[0][1] as int
+
+    }
+
+    void setupFileEventsLogging() {
```

Comment:
We don't need the file events logging, do we? This should be handled by the verbose logging already.

Reply:
Removed.